thirdeye/thirdeye-frontend/app/pods/rootcause-timeseries-cache/service.js

```diff
@@ -81,39 +77,23 @@ export default Service.extend({
 
   _extractTimeseries(json, urn) {
     const timeseries = {};
-    Object.keys(json).forEach(range =>
-      Object.keys(json[range]).filter(sid => sid != 'timestamp').forEach(sid => {
-        const jrng = json[range];
-        timeseries[urn] = {
-          timestamps: jrng.timestamp,
-          values: jrng[sid]
-        };
-      })
-    );
+    timeseries[urn] = json;
     return timeseries;
   },
 
-  _offsetTimeseries(timeseries, offset) {
-    Object.keys(timeseries).forEach(urn => {
-      timeseries[urn].timestamps = timeseries[urn].timestamps.map(t => t + offset);
-    });
-    return timeseries;
-  },
-
-  _fetchSlice(urn, range, context, offsetFunc) {
-    const metricId = urn.split(':')[3];
-    const metricFilters = toFilters([urn]);
-    const filters = toFilterMap(metricFilters);
-
-    const filterString = encodeURIComponent(JSON.stringify(filters));
+  _fetchSlice(urn, context) {
+    const metricUrn = toMetricUrn(urn);
+    const range = context.analysisRange;
+    const offset = toAbsoluteUrn(urn, context.compareMode).split(':')[2].toLowerCase();
+    const granularity = context.granularity;
+    const timezone = moment.tz.guess();
 
-    const url = `/timeseries/query?metricIds=${metricId}&ranges=${range[0]}:${range[1]}&filters=${filterString}&granularity=${context.granularity}`;
+    const url = `/rootcause/metric/timeseries?urn=${metricUrn}&start=${range[0]}&end=${range[1]}&offset=${offset}&granularity=${granularity}&timezone=${timezone}`;
     return fetch(url)
       .then(checkStatus)
       .then(res => this._extractTimeseries(res, urn))
-      .then(res => offsetFunc(res))
       .then(res => this._complete(context, res))
-      .catch(error => this._handleError(urn, error));
+      // .catch(error => this._handleError(urn, error));
```

Comment:
Any reason why we're not handling the error anymore?

Reply:
that, sir, is an excellent catch